src/mumble/AudioPreprocessor.h

```diff
@@ -0,0 +1,89 @@
+// Copyright 2024 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#ifndef MUMBLE_MUMBLE_AUDIOPREPROCESSOR_H_
+#define MUMBLE_MUMBLE_AUDIOPREPROCESSOR_H_
+
+#include <cstdint>
+#include <vector>
+
+struct SpeexEchoState_;
+struct SpeexPreprocessState_;
+
+class AudioPreprocessor {
+public:
+	/**
+	 *
+	 * \return Whether the object is initialized and its methods can be safely called.
+	 */
+	constexpr explicit operator bool() const { return m_handle != nullptr; }
+
+	AudioPreprocessor();
+	AudioPreprocessor(AudioPreprocessor &&other);
+	~AudioPreprocessor();
+
+	/**
+	 *
+	 * \param sampleRate The sample rate in Hz.
+	 *
+	 * \param quantum The number of samples that are expected for each preprocess cycle.
+	 *                Should correspond to 10-20 ms (480-960 with a sample rate of 48000).
+	 *
+	 * \return Whether initialization succeeded.
+	 */
+	bool init(std::uint32_t sampleRate, std::uint32_t quantum);
+	void deinit();
+
+	/**
+	 *
+	 * \param samples The samples that should be processed (modified in-place).
+	 *
+	 * \return Whether speech is detected in the provided samples (only if VAD is enabled).
+	 */
+	bool run(std::int16_t &samples);
+
+	SpeexEchoState_ *getEchoState();
+	bool setEchoState(SpeexEchoState_ *handle);
+
+	bool usesAGC() const;
+	bool toggleAGC(bool enable);
+	std::int32_t getAGCDecrement() const;
+	bool setAGCDecrement(std::int32_t value);
+	std::int32_t getAGCGain() const;
+	std::int32_t getAGCIncrement() const;
+	bool setAGCIncrement(std::int32_t value);
+	std::int32_t getAGCMaxGain() const;
+	bool setAGCMaxGain(std::int32_t value);
+	std::int32_t getAGCTarget() const;
+	bool setAGCTarget(std::int32_t value);
+
+	bool usesDenoise() const;
+	bool toggleDenoise(bool enable);
+
+	bool usesDereverb() const;
+	bool toggleDereverb(bool enable);
+
+	std::int32_t getNoiseSuppress() const;
+	bool setNoiseSuppress(std::int32_t value);
+
+	std::vector< std::int32_t > getPSD() const;
+	std::vector< std::int32_t > getNoisePSD() const;
+
+	std::int32_t getSpeechProb() const;
+
+	bool usesVAD() const;
+	bool toggleVAD(bool enable);
```

Comment:
I feel like all of these could use documentation - in particular those functions making use of acronyms